assets/js/modules/analytics/datastore/setup.js

```diff
@@ -0,0 +1,204 @@
+	isValidPropertyID,
+} from '../util';
+import { PROPERTY_CREATE, PROFILE_CREATE } from './constants';
+
+const { createRegistrySelector, createRegistryControl } = Data;
+
+// Actions
+const SUBMIT_CHANGES_START = 'SUBMIT_CHANGES_START';
+const SUBMIT_CHANGES_COMPLETED = 'SUBMIT_CHANGES_COMPLETED';
+const SUBMIT_PROPERTY_CREATE = 'SUBMIT_PROPERTY_CREATE';
+const SUBMIT_PROFILE_CREATE = 'SUBMIT_PROFILE_CREATE';
+const SUBMIT_SAVE_SETTINGS = 'SUBMIT_SAVE_SETTINGS';
+const SUBMIT_CHANGES_FAILED = 'SUBMIT_CHANGES_FAILED';
+
+export const INITIAL_STATE = {
+	isDoingSubmitChanges: false,
+};
+
+export const actions = {
+	*submitChanges( onSuccess = () => {} ) {
+		yield actions.startSubmitChanges();
+
+		const registry = yield Data.commonActions.getRegistry();
+		let propertyID = registry.select( STORE_NAME ).getPropertyID();
+
+		if ( propertyID === PROPERTY_CREATE ) {
+			const accountID = registry.select( STORE_NAME ).getAccountID();
+
+			const { payload } = yield actions.submitPropertyCreate( accountID );
+			const { property, error } = payload;
+
+			if ( property ) {
+				propertyID = property.id;
+			} else if ( error ) {
+				return actions.submitChangesFailed( { error } );
+			}
+		}
+
+		const profileID = registry.select( STORE_NAME ).getProfileID();
+		if ( profileID === PROFILE_CREATE ) {
+			const accountID = registry.select( STORE_NAME ).getAccountID();
+
+			const { payload } = yield actions.submitProfileCreate( accountID, propertyID );
+			const { error } = payload;
+
+			if ( error ) {
+				return actions.submitChangesFailed( { error } );
+			}
+		}
+
+		const { payload } = yield actions.submitSaveSettings();
+		if ( payload.error ) {
+			return actions.submitChangesFailed( { error: payload.error } );
+		}
+
+		onSuccess();
+
+		return actions.finishSubmitChanges();
+	},
+	submitPropertyCreate( accountID ) {
+		return {
+			payload: { accountID },
+			type: SUBMIT_PROPERTY_CREATE,
+		};
+	},
+	submitProfileCreate( accountID, propertyID ) {
+		return {
+			payload: { accountID, propertyID },
+			type: SUBMIT_PROFILE_CREATE,
+		};
+	},
+	submitSaveSettings() {
+		return { type: SUBMIT_SAVE_SETTINGS };
+	},
+	startSubmitChanges() {
+		return { type: SUBMIT_CHANGES_START };
+	},
+	finishSubmitChanges() {
+		return { type: SUBMIT_CHANGES_COMPLETED };
+	},
+	submitChangesFailed( { error } ) {
+		return {
+			payload: { error },
+			type: SUBMIT_CHANGES_FAILED,
+		};
+	},
+};
+
+export const controls = {
+	[ SUBMIT_PROPERTY_CREATE ]: createRegistryControl( ( registry ) => ( { payload } ) => {
+		return registry.dispatch( STORE_NAME ).createProperty( payload.accountID );
+	} ),
+	[ SUBMIT_PROFILE_CREATE ]: createRegistryControl( ( registry ) => ( { payload } ) => {
+		const { accountID, propertyID } = payload;
+		return registry.dispatch( STORE_NAME ).createProfile( accountID, propertyID );
+	} ),
+	[ SUBMIT_SAVE_SETTINGS ]: createRegistryControl( ( registry ) => () => {
+		return registry.dispatch( STORE_NAME ).saveSettings();
+	} ),
```

Comment:
I think these controls and related actions are overkill. I think we only need the `submitChanges` action and related internal actions. That action should be able to simply call the existing `createProperty`, `createProfile`, and `saveSettings` actions on its own.